_posts/2019-05-10-continuous-deployment-the-first-step-on-the-road-to-recovery.markdown

```diff
@@ -0,0 +1,63 @@
+> We consider origin/master to be the main branch where the source code of HEAD always reflects a production-ready state.
+> We consider origin/develop to be the main branch where the source code of HEAD always reflects a state with the latest delivered development changes for the next release.
+
+Maintaining multiple long-lived branches are an inhitibor to flow.  Changes queue on a branch, waiting for release to production.  They add complexity in understanding the divergence between branches, increasing the risk of regressions and merge hell.  This risk grows exponentially with the number of branches maintained.  Furthermore, the increased complexity of managing multiple branches, and the batching of commits, increases the complexity of a release and therefore the associated risk.  As a result, this leads to a downward spiral of increased batching, complexity and risk.
+
+Teams choose git flow for well intentioned reasons.  Faith in the codebase is low, caused by insufficient feedback on changes provided by a comprehensive automated test suite.  This leads to regular issues being found, further reducing faith.  Teams want control over feature releases: when they are confident of the quality. Batching and branching are an attempt to provide the stability to regain control.  However, it [has been shown](https://continuousdelivery.com/evidence-case-studies/) that these practices inhibit, rather than enable, flow.
+
+# Releasing the pressure
+
+A good practice for for [The First Way](https://itrevolution.com/the-three-ways-principles-underpinning-devops/) is [one piece flow](https://www.kaizenworld.com/kaizen/one-piece-flow.html).  Work on single increments of value, delivering them as soon as possible.  
+
+As Behaviour Driven Development practitioners, we like to [_start with the end in mind_](https://cucumber.io/blog/cukeup-au-2015-videos/#sharon-robson).  We consider the goal we are trying to achieve, and work backwards to how we can achieve it.  Many teams consider one piece flow as the final step in their improvement of flow.  However, many teams never get there, nor make the changes necessary in order to deliver it.  Manual approval gates are never removed; confidence in the code remains low; automated test coverage remains poor; big batch releases are common.
+
+One piece flow - a simple idea with a simple implementation: continuous deployment. Every commit flows through your continuous delivery pipeline and, assuming all conditions pass, makes its way to production.  The idea and the implementation are simple, but there are a number of fundamental principles that you must adhere to:
+ - You have a single, consistent pipeline flow of changes into production
+ - You must have confidence in your pipeline to publish changes to the next stage
+ - In order to prevent bottlenecks, you need a highliy automated pipeline
+ - In order to prevent bottlenecks, you need a high level of automated test coverage.
```

Comment:
I would probably remove the last dot point and update the third one:
>In order to prevent bottlenecks, you need a highly automated pipeline, and a high level of automated test coverage.

(also spelling of `highliy`)

Reply:
:+1: